deep_quoridor/src/train_alphazero.py

```diff
@@ -168,45 +172,41 @@ def train_alphazero(args, evaluator_server: EvaluatorServer, workers: list[Worke
                 + f"replay buffer size ({len(replay_buffer)})"
             )
 
-        # A bit of a hack.... the wandb plugin uses episode_count as the X axis for
-        # graphs, but that isn't automatically set since we aren't calling the end_game
-        # method after games in self-play (since those are run in other processes).
-        wandb_train_plugin.episode_count = epoch * args.games_per_epoch
-        save_and_test_model(training_agent, f"_epoch_{epoch}", wandb_train_plugin)
+            game_num = epoch * args.games_per_epoch
+            model_suffix = (f"_epoch_{epoch}",)
+            model_path = Path(training_agent.params.wandb_dir).mkdir(parents=True, exist_ok=True)
+            training_agent.save_model(model_path / training_agent.resolve_filename(model_suffix))
+            if wandb_train_plugin.params.upload_model:
+                # Save the model where the plugin wants it and use the plugin to compute metrics.
+                wandb_train_plugin.episode_count = game_num
+                wandb_train_plugin.compute_tournament_metrics(model_path)
 
         print(evaluator_server.get_statistics())
         for r in results:
             print(r.evaluator_statistics)
 
 
-def save_and_test_model(agent: AlphaZeroAgent, model_suffix: str, wandb_train_plugin: WandbTrainPlugin, upload=False):
-    if upload:
-        model_dir = Path(agent.params.wandb_dir)
-    else:
-        model_dir = Path(agent.params.model_dir)
-
-    # Save the model
-    Path(model_dir).mkdir(parents=True, exist_ok=True)
-    model_path = model_dir / agent.resolve_filename(model_suffix)
-    print(f"Saving checkpoint model to {model_path}")
-    agent.save_model(model_path)
-
-    # Create directory for saving models if it doesn't exist
-    wandb_train_plugin.compute_tournament_metrics(model_path)
+def setup(args) -> tuple[EvaluatorServer, list[Worker]]:
+    set_deterministic(args.seed)
 
+    if args.wandb == "":
+        wandb_params = WandbParams()
+    else:
+        wandb_params = parse_subargs(args.wandb, WandbParams)
+        assert isinstance(wandb_params, WandbParams)
```

Comment:
I tried running without `-w` (e.g. `train_alphazero.py -N 5 -W 3 -e 100`) and it hung, seems that the parsing doesn't work because args.wandb is None.   In that case, I think we shouldn't have a WandbPlugin